### PR TITLE
allow to overwrite the ssl_dir parameter

### PR DESCRIPTION
**Pull request by fraenki**
This PR makes it possible to overwrite the `ssl_dir` parameter from hiera:

```
bacula::params::ssl: true
bacula::params::ssl_dir: '/foo/bar/puppet/ssl'
```

I think this aligns well with the other hiera configurable parameters. And as a bonus, it loosens the dependency on `lpoperations/puppet` and makes it possible to use any other module to configure puppet (i.e. `theforeman/puppet`, which I strongly prefer).

This commit shouldn't break anything.

### CODE REVIEW

**Comment by xaque208** [comment]
So after thinking about this for about two minutes, and looking at the conversation in https://github.com/xaque208/puppet-bacula/pull/41, I think I'm in favor of treating the SSL dir just like the other params that are looked up from hiera above.  If folks are using PE, then they can pass in the PE SSL dir by setting the associated hiera data.  This would solve the dependency issue on ploperations/puppet and at the same time provide flexibility in SSL management for platforms that differ between Director/Storage/Client.

In short, I think this will need a bit more work to be the correct solution, and I would also like to give useful warnings/errors to users to help the upgrade path.  We can cut a major release for the module due to the dependency change and backwards incompatibility.  I'd also expect and update the the README is in order.

Can you make those adjustments here, or would you rather we break this work out into separate issues/PRs?

**Comment by fraenki** [comment]
@xaque208 I'd prefer separate issues, but I'm not sure what can be put in a separate issue.

**Comment by xaque208** [comment]
https://github.com/xaque208/puppet-bacula/issues/45

**Comment by xaque208** [comment]
Thank you for this work, though I am going to close this in favor of https://github.com/xaque208/puppet-bacula/pull/49

This implements the same interface that you have described here using hiera, but also removes the ploperations dependency.
